src/main/java/org/prebid/server/bidder/bidstack/BidstackBidder.java

```diff
@@ -0,0 +1,159 @@
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.bidstack.ExtImpBidstack;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class BidstackBidder implements Bidder<BidRequest> {
+
+    private static final String BIDDER_CURRENCY = "USD";
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    private static final TypeReference<ExtPrebid<?, ExtImpBidstack>> BIDDER_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    public BidstackBidder(String endpointUrl,
+                          CurrencyConversionService currencyConversionService,
+                          JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> updatedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final BigDecimal resolvedBidFloor = resolveBidFloor(request, imp);
+                updatedImps.add(updateImp(imp, resolvedBidFloor));
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest updatedBidRequest = updateBidRequest(request, updatedImps);
+
+        return Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(constructHeaders(updatedBidRequest))
+                .body(mapper.encodeToBytes(updatedBidRequest))
+                .payload(updatedBidRequest)
+                .build());
+    }
+
+    private ExtImpBidstack parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper()
+                    .convertValue(imp.getExt(), BIDDER_EXT_TYPE_REFERENCE)
+                    .getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Unable to decode the impression ext for id: " + imp.getId());
+        }
+    }
+
+    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
+        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
```

Comment:
What about to move `bidBloor` resolving to `imp` update stage? It will allow us to return same `imp` object in case when nothing changed.

```
updateImp(...) {
    if (!shouldConvertBidFloor(...)) {
          // return original imp
    }

    // resolve bidFloor and update imp
}
```

Reply:
I have done this, but to avoid PreBidException must also add `parseExtImp(imp);`. Can this case really happen?
```
        final BidRequest bidRequest = givenBidRequest(impCustomizer -> impCustomizer
                .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))));
```

Reply:
I have rollback this solution because it also has some additional tables. We need to make sure that imp has `bidfloorcur` and this force to use check or still have imp build if `bidfloorcur`. To make sure that in response included `imp.bidfloorcur`

Reply:
What about this solution?
```
    private Imp updateImp(Imp imp, BidRequest request) {
        final UpdateResult<BigDecimal> resolvedBidFloor = resolveBidFloor(request, imp);

        return resolvedBidFloor.isUpdated()
                ? imp.toBuilder()
                .bidfloorcur(BIDDER_CURRENCY)
                .bidfloor(resolvedBidFloor.getValue())
                .build()
                : imp;
    }

    private UpdateResult<BigDecimal> resolveBidFloor(BidRequest request, Imp imp) {
        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
                ? UpdateResult.updated(convertBidFloorCurrency(
                imp.getBidfloor(), request, imp.getId(), imp.getBidfloorcur()))
                : UpdateResult.unaltered(imp.getBidfloor());
    }
```